v2/tools/generator/internal/codegen/testdata/ArmResource/Arm_test_dependent_resource_and_ownership_azure.golden

```diff
@@ -193,7 +193,32 @@ func (a *A) updateValidations() []func(old runtime.Object) error {
 		func(old runtime.Object) error {
 			return a.validateResourceReferences()
 		},
+		a.validateImmutableProperties}
+}
+
+// validateImmutableProperties validates all immutable properties
+func (a *A) validateImmutableProperties(old runtime.Object) error {
+
+	resourceID := genruntime.GetResourceIDOrDefault(a)
```

Comment:
rather than inlining the code here, can you generate a call to a function in `genruntime` instead? we should be able to reuse the same code for all resource types, which will save on code size (binary and textual)

Reply:
Done!